controllers/api/v1/search_test.go

```diff
@@ -0,0 +1,318 @@
+package apiv1
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/google/uuid"
+
+	_ "google-scraper/initializers"
+	. "google-scraper/tests"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("SearchController", func() {
+	Describe("POST /api/v1/search", func() {
+		Context("Given the valid credential", func() {
+			Context("given the params are valid", func() {
+				It("returns 201 status code", func() {
+					user := FabricateUser("John", "john@example.com", "secret")
+					client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+					oauthToken := FabricateOAuthToken(client, user.Id)
+					filePath := AppRootDir() + "/tests/fixtures/shared/valid_keywords.csv"
+					_, body := CreateMultipartFormData(filePath)
+					mockResponseFilePath := AppRootDir() + "/tests/fixtures/services/crawler/valid_get_response.html"
+					MockCrawling(mockResponseFilePath)
+
+					header := http.Header{
+						"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())},
+						"Content-Type":  {"multipart/form-data; boundary=multipart-boundary"}}
+
+					response := MakeAuthenticatedRequest("POST", "/api/v1/search", header, body, nil)
+
+					Expect(response.StatusCode).To(Equal(http.StatusCreated))
+				})
+
+				It("returns empty response", func() {
+					user := FabricateUser("John", "john@example.com", "secret")
+					client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+					oauthToken := FabricateOAuthToken(client, user.Id)
+					filePath := AppRootDir() + "/tests/fixtures/shared/valid_keywords.csv"
+					_, body := CreateMultipartFormData(filePath)
+
+					header := http.Header{
+						"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())},
+						"Content-Type":  {"multipart/form-data; boundary=multipart-boundary"}}
+
+					response := MakeAuthenticatedRequest("POST", "/api/v1/search", header, body, nil)
+					responseBody, err := ioutil.ReadAll(response.Body)
+					if err != nil {
+						Fail("Reading response body failed: " + err.Error())
+					}
+
+					Expect(string(responseBody)).To(BeEmpty())
+				})
+			})
+
+			Context("given the params are INVALID", func() {
+				Context("given the uploaded file is nil", func() {
+					It("returns 401 status code", func() {
```

Comment:
Did you mean `422` as it returns `StatusUnprocessableEntity`? `401` would mean an authorization issue which should not happen in this case.

Reply:
Yes, I meant `422`. Updated it also.